src/Watcher/Resources/ItemIndexDocumentBuilder.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\elasticsearch\Watcher\Resources;
+
+use oat\tao\model\search\index\IndexDocument;
+use oat\taoQtiItem\model\qti\Service;
+
+class ItemIndexDocumentBuilder extends AbstractIndexDocumentBuilder
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function createDocumentFromResource(\core_kernel_classes_Resource $resource): ?IndexDocument
```

Comment:
As I can see, the only change from the method on `GenericIndexDocumentBuilder` is the body, maybe body can be a parameter on the method and then they don't need to be different?

The same for the other types

Reply:
Can be, but aren't we loosing BC in that case? Because as I see previous implementations are using only Resource/array (depending on method) as a param. That's actually why in `IndexService` class in other PR, both methods are marked as deprecated. Later on, when we will consider that we can get rid of those two deprecateds we can also change signatures.
How does that sound for you?

Reply:
Seems fine for me, thanks for the explanation